### PR TITLE
Backup job support specify priorityClassName.

### LINKED ISSUE #4052: Support PriorityClass on Backup jobs
## Feature Request

**Is your feature request related to a problem? Please describe:**
Without a priorityClass, the backup pods could be evicted during the backup and cause the backup process to fail.

**Describe the feature you'd like:**
Should either allow the user to specify a priorityClass or default a higher priorityClass to the backup jobs.

**Describe alternatives you've considered:**
Add in the documentation saying that this a possible failure, and add diagnosis and mitigation steps.

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by mikechengwei**
### What problem does this PR solve?
Fix #4052 .

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Backup job support specify priorityClassName.
```

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 1]
```diff
@@ -1416,6 +1416,11 @@ type BackupSpec struct {
 	// PodSecurityContext of the component
 	// +optional
 	PodSecurityContext *corev1.PodSecurityContext `json:"podSecurityContext,omitempty"`
+
+	// PriorityClassName of the component. Override the cluster-level one if present
+	// Optional: Defaults to cluster-level setting
+	// +optional
+	PriorityClassName *string `json:"priorityClassName,omitempty"`
```
Comment (DanielZhangQD) [inline pkg/apis/pingcap/v1alpha1/types.go:1423]: Please also cover restore.

File: pkg/backup/backup/backup_manager.go [hunk 2]
```diff
@@ -325,7 +325,9 @@ func (bm *backupManager) makeExportJob(backup *v1alpha1.Backup) (*batchv1.Job, s
 			Template:     *podSpec,
 		},
 	}
-
+	if backup.Spec.PriorityClassName != nil {
```
Comment (DanielZhangQD) [inline pkg/backup/backup/backup_manager.go:328]: Please also cover the backup, restore, import, and clean job.

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 3]
```diff
@@ -1693,6 +1698,11 @@ type RestoreSpec struct {
 	// PodSecurityContext of the component
 	// +optional
 	PodSecurityContext *corev1.PodSecurityContext `json:"podSecurityContext,omitempty"`
+
+	// PriorityClassName of Restore Job Pods
+	// Optional: Defaults to omitted
+	// +optional
+	PriorityClassName *string `json:"priorityClassName,omitempty"`
```
Comment (DanielZhangQD) [inline pkg/apis/pingcap/v1alpha1/types.go:1705]: No need to be a pointer, just `string` is good, WDYT?

File: pkg/backup/backup/backup_manager.go [hunk 4]
```diff
@@ -325,7 +325,9 @@ func (bm *backupManager) makeExportJob(backup *v1alpha1.Backup) (*batchv1.Job, s
...
+		job.Spec.Template.Spec.PriorityClassName = *backup.Spec.PriorityClassName
```
Comment (DanielZhangQD) [inline pkg/backup/backup/backup_manager.go:329]: Please also update the backup job.

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Comment by mikechengwei** [comment]
LGTM

**Review by csuzhangxc — APPROVED** [review]

**Pull request merged**